Toolkit/ArcGISToolkit/UnitsViewController.swift

```diff
@@ -125,155 +110,104 @@ public class UnitsViewController: TableViewController, UINavigationBarDelegate,
         
-        if NSLocale.current.usesMetricSystem{
-            selectedLinearUnit = AGSLinearUnit.kilometers()
-            selectedAreaUnit = AGSAreaUnit(unitID: AGSAreaUnitID.hectares) ?? AGSAreaUnit.squareKilometers()
-        }
-        else{
-            selectedLinearUnit = AGSLinearUnit.miles()
-            selectedAreaUnit = AGSAreaUnit(unitID: AGSAreaUnitID.acres) ?? AGSAreaUnit.squareMiles()
-        }
-        
-        selectedAngularUnit = AGSAngularUnit.degrees()
+        return searchController
     }
     
-    
     override public func viewDidLoad() {
         super.viewDidLoad()
         
-        // setup navbar
-        let navbar = UINavigationBar(frame: CGRect(x: 0, y: 0, width: self.view.bounds.size.width, height: 64))
-        navbar.autoresizingMask = .flexibleWidth
-        view.addSubview(navbar)
-        navbar.delegate = self
-        
-        let cancelButton = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(cancelAction as ()->Void))
-        
-        let item = UINavigationItem(title: "Units")
-        item.leftBarButtonItem = cancelButton
-        navbar.pushItem(item, animated: false)
-        
-        //
-        let insets = UIEdgeInsets(top: navbar.bounds.size.height, left: 0, bottom: 0, right: 0)
-        tableView.contentInset = insets
-        tableView.scrollIndicatorInsets = insets
-        tableView.scrollRectToVisible(CGRect(x: 0, y: 0, width: 1, height: 1), animated: false)
-        
-        // search bar
-        let searchbar = UISearchBar(frame: CGRect(x: 0, y: 0, width: self.view.bounds.size.width, height: 44))
-        searchbar.delegate = self
-        searchbar.spellCheckingType = .no
-        searchbar.autocapitalizationType = .none
-        searchbar.autocorrectionType = .no
-        tableView.tableHeaderView = searchbar
-        
-    }
-    
-    // MARK: SearchBar / NavBar delegates
-    
-    public func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        if searchText.isEmpty{
-            self.currentUnits = self.unfilteredUnits
+        if #available(iOS 11.0, *) {
+            // Nothing to do!
+        } else {
+            tableView.tableHeaderView = searchController?.searchBar
         }
-        else{
-            
-            DispatchQueue.global(qos: DispatchQoS.QoSClass.userInitiated).async {
-                let filtered = self.unfilteredUnits.filter{
-                    $0.pluralDisplayName.range(of: searchText, options: .caseInsensitive) != nil
-                }
-                DispatchQueue.main.async {
-                    self.currentUnits = filtered
-                }
-            }
-            
-        }
-    }
-    
-    public func position(for bar: UIBarPositioning) -> UIBarPosition {
-        return .topAttached
     }
     
     // MARK: TableView delegate/datasource methods
     
-    func tableView(_ tableView: UITableView, didSelectRowAtIndexPath indexPath: IndexPath) {
```

Comment:
This method had the wrong signature, so it was never getting called. Selecting a unit had not effect.